packages/brisa/src/utils/rpc/rpc-integration.test.ts

```diff
@@ -437,6 +437,11 @@ describe('utils', () => {
       window._xm = null;
     });
 
+    it('should not full reload when the destination URL is the same as the current location', async () => {
+      await simulateSPANavigation('http://localhost');
+      expect(mockNavigationIntercept).toHaveBeenCalled();
```

Comment:
`mockNavitationIntercept` should not be called now, you miss `location.href = 'https://localhost'` before simulate the SPA navigation to reproduce the condition. Feel free also to adapt the simulator to add the `preventDefault` mock and check here that the `preventDefault` is called.

To run these tests you can run this:


```sh
bun test packages/brisa/src/utils/rpc/rpc-integration.test.ts --watch
```